File: clojure/src/spoonacular_api/specs/search_recipes_by_ingredients_200_response_inner_missed_ingredients_inner.clj
(ns spoonacular-api.specs.search-recipes-by-ingredients-200-response-inner-missed-ingredients-inner
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def search-recipes-by-ingredients-200-response-inner-missed-ingredients-inner-data
  {
   (ds/req :aisle) string?
   (ds/req :amount) float?
   (ds/req :id) int?
   (ds/req :image) string?
   (ds/opt :meta) (s/coll-of string?)
   (ds/req :name) string?
   (ds/opt :extendedName) string?
   (ds/req :original) string?
   (ds/req :originalName) string?
   (ds/req :unit) string?
   (ds/req :unitLong) string?
   (ds/req :unitShort) string?
   })

(def search-recipes-by-ingredients-200-response-inner-missed-ingredients-inner-spec
  (ds/spec
    {:name ::search-recipes-by-ingredients-200-response-inner-missed-ingredients-inner
     :spec search-recipes-by-ingredients-200-response-inner-missed-ingredients-inner-data}))

File: scala/src/main/scala/org/openapitools/models/SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import org.openapitools.models.BigDecimal
import scala.collection.immutable.Seq

/**
 * 
 * @param aisle 
 * @param amount 
 * @param id 
 * @param image 
 * @param meta 
 * @param name 
 * @param extendedName 
 * @param original 
 * @param originalName 
 * @param unit 
 * @param unitLong 
 * @param unitShort 
 */
case class SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner(aisle: String,
                amount: BigDecimal,
                id: Int,
                image: String,
                meta: Option[Seq[String]],
                name: String,
                extendedName: Option[String],
                original: String,
                originalName: String,
                unit: String,
                unitLong: String,
                unitShort: String
                )

object SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner {
    /**
     * Creates the codec for converting SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner from and to JSON.
     */
    implicit val decoder: Decoder[SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner] = deriveDecoder
    implicit val encoder: ObjectEncoder[SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner] = deriveEncoder
}
